### PR TITLE
Improve YuvToRgbConverter.kt

### PR DESCRIPTION
**Pull request by rogerthat94**
The YuvToRgbConverter stores the NV21 Byte array in a ByteBuffer. However, none of the ByteBuffer features are ever used. Every time the ByteBuffer is accessed or modified, the code just calls the `.array()` method. 

It seems better to just use a `ByteArray` instead of a `ByteBuffer`. That's what this change does. 

I see there are [four other copies](https://github.com/android/camera-samples/search?q=YuvToRgbConverter&unscoped_q=YuvToRgbConverter) of this file in the repository. However, I decided to just make this modification once, since I'm just proposing these changes. I would be happy to copy these changes over to the other copies of this file, if this pull request gets approved. However, this appears to be the only copy of this file that gets used.

### CODE REVIEW

**Comment by owahltinez** [comment]
Thank you for this PR!

> I see there are four other copies of this file in the repository

Sorry, it is a little confusing and not well documented, but we must keep all folders within this repo "independent" which is why there is a lot of duplicity. Kindly follow the following steps to help us evaluate this change:
1. Re-apply your change to this file: https://github.com/android/camera-samples/blob/master/CameraUtils/lib/src/main/java/com/example/android/camera/utils/YuvToRgbConverter.kt
2. Run `sh .github/scripts/copy_utils.sh` from the root of the Github repo
3. Update this pull request

**Comment by rogerthat94** [comment]
No problem! 

Thanks for the prompt reply about this. I have followed your instructions. Please let there is anything else that you want to to do to get this pull request reviewed.

**Comment by owahltinez** [comment]
Thank you! This seems like a straightforward change. What devices did you test this with? Did you observe a noticeable difference in the performance?

**Comment by rogerthat94** [comment]
I tested this on an Essential PH-1. The performance seemed about the same, but I only tested with a single resolution.

File: Camera2Basic/utils/src/main/java/com/example/android/camera/utils/YuvToRgbConverter.kt [hunk 1]
```diff
@@ -59,33 +58,30 @@ class YuvToRgbConverter(context: Context) {
...
             val pixelSizeBits = ImageFormat.getBitsPerPixel(ImageFormat.YUV_420_888)
-            yuvBuffer = ByteBuffer.allocateDirect(pixelCount * pixelSizeBits / 8)
+            yuvBuffer = ByteArray(pixelCount * pixelSizeBits / 8)
         }
 
-        // Rewind the buffer; no need to clear it since it will be filled
-        yuvBuffer.rewind()
-
         // Get the YUV data in byte array form using NV21 format
-        imageToByteBuffer(image, yuvBuffer.array())
```
Comment (alexcohn) [inline Camera2Basic/utils/src/main/java/com/example/android/camera/utils/YuvToRgbConverter.kt:69]: This was a dangerous line: direct ByteBuffer is not required to be backed by array!

Comment (rogerthat94) [reply Camera2Basic/utils/src/main/java/com/example/android/camera/utils/YuvToRgbConverter.kt:69]: Good point. 

For others: see [here](https://developer.android.com/reference/java/nio/ByteBuffer.html#array()) for exceptions this method can throw.

**Review by owahltinez — APPROVED** [review]
Sorry for the delay reviewing this

**Comment by rogerthat94** [comment]
No problem! Thank you for merging this.

**Comment by alexcohn** [comment]
> we must keep all folders within this repo "independent"

why not use a common **utils** directory?

**Pull request merged**